server/metrics/metrics.go

```diff
@@ -251,6 +251,40 @@ var (
 	/// sum(rate(buildbuddy_remote_execution_count[5m]))
 	/// ```
 
+	RemoteExecutionQueueDurationUsec = promauto.NewHistogram(prometheus.HistogramOpts{
+		Namespace: bbNamespace,
+		Subsystem: "remote_execution",
+		Name:      "queue_duration_usec",
+		Help:      "Time each action spends waiting in the queue, in **microseconds**.",
+	})
+
+	/// #### Examples
+	///
+	/// ```promql
+	/// # Median action queue duration
+	/// histogram_quantile(
+	///	  0.5,
+	///   sum(rate(buildbuddy_remote_execution_queue_duration_usec_bucket[5m])) by (le)
+	/// )
+	/// ```
+
+	RemoteExecutionCommandDurationUsec = promauto.NewHistogram(prometheus.HistogramOpts{
```

Comment:
you can tell me if this doesn't make sense, but i know in executed action metadata we have 5 durations -- what if you had a single Histogram and added a label that could be one of those 5 things, then you could record them all. WDYT?

Reply:
So far I've been using labels for "filtering" a particular metric where the values are all measuring the same thing. It has the benefit that it prevents accidentally doing aggregations across different metrics which don't make any sense (e.g. averaging the durations from different command processing phases).

But using labels as a way to break down the "phases" of a duration metric also makes sense to me -- it has the nice benefit that it allows adding more easily sub-phases to get a more granular performance breakdown, without needing to add new metrics every time.

IIUC either approach winds up creating a new time series, and are probably both approximately the same cost in terms of data storage.

So in this case I think it makes sense to use labels (done), since we will probably want to make sure we track each phase of execution so that we have a solid understanding of where the bottlenecks are.